test/utils/binarySearch.js

```diff
@@ -32,13 +32,11 @@ tap.test('binary search', t => {
     { input: [l4, 'cde'], expected: 6, message: 'find item with similarity in n=odd array' }
   ]
 
-  t.plan(tests.length + 1)
+  t.plan(tests.length)
 
   tests.forEach(({ input: [list, target], expected, message }) => {
     t.equal(expected, binarySearch(list, target), message)
   })
-
-  t.throws(() => binarySearch(['a'], 'a'), Error('List length must be even'))
```

Comment:
Why are we removing this test? The custom binary search algorithm needs to support the even-lengthed nature of the header list

Reply:
It's only used internally and should never happen. Better to add an assertion if you want a check.

Reply:
like an assert in the binarySearch method itself? I'm +1 for that so folks don't miss-use this in the future